crates/ruff_linter/src/registry/rule_set.rs

```diff
@@ -243,6 +244,20 @@ impl RuleSet {
         self.0[index] & mask != 0
     }
 
+    /// Returns `true` if any of the rules in `rules` are in this set.
+    #[inline]
+    pub const fn any(&self, rules: &[Rule]) -> bool {
+        let mut any = false;
+        let mut i = 0;
+
+        while i < rules.len() {
+            any |= self.contains(rules[i]);
```

Comment:
I intentionally use an bitwise OR here to avoid any branching.

Reply:
I'd bet it would compile down to the same code if you used an `if`. :-) But I actually find this pretty readable as it is personally.

Reply:
Acutally, it does not

The iter version has a jump

```
mov     rax, qword ptr [rsp - 112]
movzx   edx, byte ptr [rsp - 116]
mov     cl, 1
bt      rax, rdx
jb      .LBB0_2
movzx   ecx, byte ptr [rsp - 114]
bt      rax, rcx
setb    cl
```

The loop version does not (but it requires more instructions)

```
mov     byte ptr [rsp - 117], cl
lea     rax, [rsp - 117]
movzx   ecx, byte ptr [rsp - 114]
mov     edx, 1
shl     edx, cl
movzx   ecx, byte ptr [rsp - 116]
bts     edx, ecx
test    dword ptr [rsp - 112], edx
setne   byte ptr [rsp - 117]
```
https://godbolt.org/z/oWWzqc6s1

Reply:
Yeah I just meant, `if self.contains(rules[i]) { any = true }`.

It makes sense that the `iter` version has an extra jump because of the short circuit.

Reply:
ah yeah, that probably compiles down to the same